zio-http/src/main/scala/zio/http/socket/SocketApp.scala

```diff
@@ -1,14 +1,15 @@
 package zio.http.socket
 
-import zio._
+import zio.{ZIO, _}
 import zio.http._
 import zio.http.model.Headers
+import zio.http.socket.SocketApp.{SocketAppHandler, SocketHandler}
 import zio.stacktracer.TracingImplicits.disableAutoTrace // scalafix:ok;
 
 final case class SocketApp[-R](
   decoder: SocketDecoder = SocketDecoder.default,
   protocol: SocketProtocol = SocketProtocol.default,
-  message: Option[ChannelEvent[WebSocketFrame, WebSocketFrame] => ZIO[R, Throwable, Any]] = None,
+  message: SocketAppEvent => Option[ZIO[R, Throwable, SocketAppAction]] = SocketAppHandler.noOp.lift
```

Comment:
What if I want to perform a side-effect before deciding what socket action to take?

Reply:
@tusharmath this is basically a lifted `PartialFunction[SocketAppEvent, ZIO[R, Throwable, SocketAppAction]]` - I wanted the user API to accept a `PartialFunction` but found the lifted version easier to work with internally.
You there is no pure match that can decide on whether to handle an event, your effect can always evaluate to `SocketAppAction.NoOp`
To tell the truth, I'm not at all sure about this encoding - was just a rough sketch that I wanted to share with @swoogles